sklearn/linear_model/_glm/glm.py

```diff
@@ -20,14 +24,563 @@
     HalfTweedieLossIdentity,
 )
 from ...base import BaseEstimator, RegressorMixin
-from ...utils.optimize import _check_optimize_result
+from ...exceptions import ConvergenceWarning
 from ...utils import check_array, deprecated
 from ...utils.validation import check_is_fitted, _check_sample_weight
-from ...utils._param_validation import Interval, StrOptions
+from ...utils._param_validation import Interval, StrOptions, Hidden
 from ...utils._openmp_helpers import _openmp_effective_n_threads
+from ...utils.optimize import _check_optimize_result
 from .._linear_loss import LinearModelLoss
 
 
+class _NewtonSolver(ABC):
```

Comment:
As eventually several solvers extending this one will be added, would it make sense to have a dedicated file for GLM solvers?

Reply:
I would not mind. @ogrisel ping.

Reply:
If it's in its own file starting with an underscore, I could remove the underscore on the class name again, right?

Reply:
Ok for a dedicated file, this file is becoming a annoying to scroll around.

Reply:
[Done in](https://github.com/scikit-learn/scikit-learn/pull/24637/commits/f002eb78a2fdb36eee4ab8ec77df0e40a2790623)